### ISSUE TITLE
./boinc: symbol lookup error: ./boinc: undefined symbol: gzopen64

### ISSUE THREAD

**Issue by romw** [BOINC/boinc-ng#516]
**Reported by jgxenite on 25 Feb 38028786 03:33 UTC**
I recently noticed that Boinc 5.10.28 had stopped running on my computer. I can find no reason for this, except for this error message that appears when running the client:

./boinc: symbol lookup error: ./boinc: undefined symbol: gzopen64

It has been doing this now for approximately two days. I found posts on various forums relating this error to libxml, but I have not updated my libxml packages (version 2.6.30-1 under Fedora 8) since installing Fedora 8 about a month ago.

**Comment by romw** [comment]
**Commented by Nicolas on 22 Jul 38028799 23:33 UTC**
Lowering priority, just because I have never seen this reported before.

**Comment by romw** [comment]
**Commented by surface on 28 Dec 38057881 04:26 UTC**
This is happening on my Gentoo install as well when BOINC tries to upload a result:

BOINC client version 5.10.28 for i686-pc-linux-gnu[libcurl/7.17.1 OpenSSL/0.9.8g zlib/1.2.3[[BR]([BR]]
Libraries:)]
OS: Linux: 2.6.23-gentoo-r3[symbol lookup error: ./boinc: undefined symbol: gzopen64[[BR]([BR]]
./boinc:)]

#ldd boinc:[=>  (0xb7fd4000)[[BR]([BR]]
linux-gate.so.1)]
libnsl.so.1 => /lib/libnsl.so.1 (0xb7fa9000)[=> /lib/libdl.so.2 (0xb7fa5000)[[BR]([BR]]
libdl.so.2)]
libz.so.1 => /lib/libz.so.1 (0xb7f93000)[=> /usr/lib/gcc/i686-pc-linux-gnu/4.1.2/libstdc++.so.6 (0xb7eb2000)[[BR]([BR]]
libstdc++.so.6)]
libpthread.so.0 => /lib/libpthread.so.0 (0xb7e9b000)[=> /lib/libm.so.6 (0xb7e76000)[[BR]([BR]]
libm.so.6)]
libc.so.6 => /lib/libc.so.6 (0xb7d4c000)[(0xb7fd5000)[[BR]([BR]]
/lib/ld-linux.so.2)]
libgcc_s.so.1 => /usr/lib/gcc/i686-pc-linux-gnu/4.1.2/libgcc_s.so.1 (0xb7d41000)[have sys-libs/zlib-1.2.3-r1[[BR]([BR]]


I)]
which does not appear to contain gzopen64:[-T /lib/libz.so.1 | grep open[[BR]([BR]]
#objdump)]
00000000      DF *UND*  00000251  GLIBC_2.1   fdopen[g    DF .text  00000012  Base        gzopen[[BR]([BR]]
000034d6)]
0000348d g    DF .text  00000049  Base        gzdopen[     DF *UND*  00000033  GLIBC_2.1   fopen[[BR]([BR]]
00000000)]


I updated the BOINC client from 5.10.21, which was working fine for me (but manager was not working)

**Comment by romw** [comment]
**Commented by nstickney on 15 Mar 38102285 00:00 UTC**
I have the same problem when uploading results:

./boinc: symbol lookup error: ./boinc: undefined symbol: gzopen64

#uname -a: [edited]
   Linux 2.6.23.14-107.fc8 #1 SMP 2008 i686 athlon i386 GNU/Linux

#ldd /opt/BOINC/boinc:
   linux-gate.so.1 =>  (0x00110000)
   libnsl.so.1 => /lib/libnsl.so.1 (0x0069f000)
   libdl.so.2 => /lib/libdl.so.2 (0x00839000)
   libz.so.1 => /lib/libz.so.1 (0x00886000)
   libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00cb6000)
   libpthread.so.0 => /lib/libpthread.so.0 (0x0086b000)
   libm.so.6 => /lib/libm.so.6 (0x00840000)
   libc.so.6 => /lib/libc.so.6 (0x006de000)
   /lib/ld-linux.so.2 (0x006bf000)
   libgcc_s.so.1 => /lib/libgcc_s.so.1 (0x00c9b000)

#objdump -T /lib/libz.so.1 | grep open:
   00000000      DF *UND*  000002fa  GLIBC_2.1   fdopen
   00000000      DF *UND*  00000032  GLIBC_2.1   fopen
   008897e0 g    DF .text  00000088  Base        gzdopen
   00889870 g    DF .text  00000014  Base        gzopen

zlib is version 1.2.3-14.fc8 from the Fedora repositories.

**Comment by romw** [comment]
**Commented by Ageless on 25 Nov 38102784 13:46 UTC**
And one more, reported on [Cosmology@Home](http://www.cosmologyathome.org/forum_thread.php?id=378):
```
>> Beginning Phase 3 <<
>> ----------------- <<

at z =   0.0000000E+00  sigma8 (all matter)=  0.8138058
06-Feb-2008 13:42:13 [Computation for task wu_013108_090634_0_2 finished
./boinc: symbol lookup error: ./boinc: undefined symbol: gzopen64

[1](Cosmology@Home])+  Exit 127                ./boinc

After restarting the client, the following message is repeated:

06-Feb-2008 13:42:29 [Restarting task wu_013108_090634_0_2 using camb version 205
06-Feb-2008 13:42:31 [Cosmology@Home](Cosmology@Home]) Task wu_013108_090634_0_2 exited with zero status but no 'finished' file
06-Feb-2008 13:42:31 [Cosmology@Home] If this happens repeatedly you may need to reset the project.
```
Returning priority to Major.

**Comment by romw** [comment]
**Commented by MichaelBuckingham on 11 Aug 38148239 07:33 UTC**

I downloaded the source for 5.10.42, compiled it and no longer have this issue.

**Comment by romw** [comment]
**Commented by MichaelBuckingham on 16 Aug 38148274 16:00 UTC**

I think whats happening is a zlib issue...zlib.h converts gzopen over to *64 calls when _FILE_OFFSET_BITS is 64 but cant define the prototypes unless _LARGEFILE64_SOURCE is also defined...incorrectly calling gzopen64.

Not sure it's a BOINC issue...??

ldd boinc
        linux-gate.so.1 =>  (0xffffe000)
        libnsl.so.1 => /lib/libnsl.so.1 (0xb7f96000)
        libcurl.so.4 => /usr/lib/libcurl.so.4 (0xb7f5e000)
        libidn.so.11 => /usr/lib/libidn.so.11 (0xb7f2c000)
        libssl.so.0.9.8 => /usr/lib/libssl.so.0.9.8 (0xb7ee9000)
        libcrypto.so.0.9.8 => /usr/lib/libcrypto.so.0.9.8 (0xb7da3000)
        libdl.so.2 => /lib/libdl.so.2 (0xb7d9f000)
        libz.so.1 => /lib/libz.so.1 (0xb7d8b000)
        libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0xb7c9d000)
        libpthread.so.0 => /lib/libpthread.so.0 (0xb7c86000)
        libm.so.6 => /lib/libm.so.6 (0xb7c61000)
        libc.so.6 => /lib/libc.so.6 (0xb7b2e000)
        /lib/ld-linux.so.2 (0xb7fd3000)
        libgcc_s.so.1 => /lib/libgcc_s.so.1 (0xb7b21000)


objdump -T /lib/libz.so.1 | grep open
00000000      DF *UND*  00000281  GLIBC_2.1   fdopen
00000000      DF *UND*  00000032  GLIBC_2.1   fopen
00003a20 g    DF .text  00000088  Base        gzdopen
00003ab0 g    DF .text  00000014  Base        gzopen

**Comment by romw** [comment]
**Commented by korpela on 8 Nov 38502128 20:00 UTC**
Should be fully fixed in check-in 15530.

**Comment by romw** [comment]
**Commented by Nicolas on 13 Jan 38502196 20:53 UTC**
Fixed in [15530].

Looks like Trac only looks for "fix !#1234" or "fixes !#1234", not "fixes ticket !#1234".

**Comment by romw** [comment]
**Commented by romw on 9 Dec 38528928 22:13 UTC**
should be fixed in latest 6.2 release
